controllers/configuration/backend/kubernetes.go

```diff
@@ -0,0 +1,126 @@
+package backend
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/oam-dev/terraform-controller/api/v1beta2"
+	"github.com/oam-dev/terraform-controller/controllers/util"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	terraformWorkspace = "default"
+	// TerraformStateNameInSecret is the key name to store Terraform state
+	TerraformStateNameInSecret = "tfstate"
+	// TFBackendSecret is the Secret name for Kubernetes backend
+	TFBackendSecret = "tfstate-%s-%s"
+)
+
+type K8SBackend struct {
+	Client       client.Client
+	HCLCode      string
+	SecretSuffix string
+	SecretNS     string
+}
+
+func getTerraformBackendSecretNS() string {
+	ns := os.Getenv("TERRAFORM_BACKEND_NAMESPACE")
+	if ns == "" {
+		ns = "vela-system"
+	}
+	return ns
+}
+
+func newDefaultK8SBackend(suffix string, client client.Client) *K8SBackend {
+	ns := getTerraformBackendSecretNS()
+	hcl := renderK8SBackendHCL(suffix, ns)
+	return &K8SBackend{
+		Client:       client,
+		HCLCode:      hcl,
+		SecretSuffix: suffix,
+		SecretNS:     ns,
+	}
+}
+
+func newK8SBackendFromHCL(hclCode string, backendConfig *ParsedBackendConfig, client client.Client) (Backend, error) {
```

Comment:
What's the meaning of `new` here?

Reply:
It means that the function will create a new instance of `K8Sbackend`.